File: 283.moveZeroes.js
/*
 * @lc app=leetcode id=283 lang=javascript
 *
 * [283] Move Zeroes
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
  const len = nums.length;
  let [nonZeroIndex, zeroIndex] = [0, 0];
  while (nonZeroIndex < len) {
    if (nums[nonZeroIndex] != 0) {
      const temp = nums[zeroIndex];
      nums[zeroIndex] = nums[nonZeroIndex];
      nums[nonZeroIndex] = temp;
      zeroIndex += 1;
    }
    nonZeroIndex += 1;
  }
};
// @lc code=end

File: 283.moveZeroes.py
#
# @lc app=leetcode id=283 lang=python3
#
# [283] Move Zeroes
#

# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zero_pointer, nonzero_pointer = 0,0
        while nonzero_pointer < len(nums):
            if nums[nonzero_pointer] != 0:
                nums[nonzero_pointer], nums[zero_pointer]= nums[zero_pointer], nums[nonzero_pointer]
                zero_pointer +=1

            nonzero_pointer += 1
        

        
# @lc code=end

 